front-end/src/components/Teacher/index.jsx

```diff
@@ -27,55 +39,133 @@ const Teacher = ({ fullName, activity, id, overview, courses }) => {
     { title: 'Overview', link: `/app/teachers/${id}/overview` },
     { title: 'Courses', link: `/app/teachers/${id}/courses` },
   ];
+  const { currentUser } = useContext(CurrentUserContext);
+  const [subscriptions, setSubscriptions] = useState([]);
+  const [isSubscripted, setIsSubscripted] = useState(false);
+  const [isLoader, setIsLoader] = useState(false);
+  const [buttonLoader, setButtonLoader] = useState(false);
+
+  const patchSubscription = async () => {
+    try {
+      setButtonLoader(true);
+      const subscribe = await createSubscription(id, currentUser.roleId);
+      await fetchStudentSubscriptions(currentUser.roleId);
+      setButtonLoader(false);
+      return subscribe;
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const fetchStudentSubscriptions = async (userId) => {
+    try {
+      setIsLoader(true);
+      const fetchedSubscriptions = await getStudentSubscription(userId);
+      setSubscriptions(fetchedSubscriptions);
+      setIsLoader(false);
+      return fetchedSubscriptions;
```

Comment:
If you don't use returning data, you can avoid `return`

Reply:
I thought that function should return data